cl/settings/10-public.py

```diff
@@ -156,6 +156,11 @@
     'person': SOLR_PEOPLE_TEST_URL,
     'recap': SOLR_RECAP_TEST_URL,
 }
+SOLR_EXAMPLE_CORE_PATH = os.path.join(os.sep, 'usr', 'local', 'solr',
```

Comment:
I have to confess that after reading this a few times, I'm still confused about local vs remote. Maybe these variable names need to be tweaked, but:

 - Why do we need an extra variable here that we didn't need before? 

 - Local/remote doesn't feel right to me. Maybe it's existing/new or prod/test...or something. But in my case, everything is local, for example.

Reply:
`LOCAL` refers to the path from the view of the machine that runs Django. `REMOTE` is the path on the Solr server. In the old setup where Solr and Django runs on the same machine, both paths are identical. However, when using Docker they need to be different, such that Solr can access it. For example, I use the following settings locally:

```python
# 15-private.py

# I copied the default Solr config to this path on my machine
SOLR_EXAMPLE_CORE_PATH = '/Volumes/data/repo/freelaw/recsys/conf/solr/cores/collection1'

# This path is mounted as volume to the Solr container
SOLR_TEMP_CORE_PATH_LOCAL = '/Volumes/data/repo/freelaw/recsys/solr2/solr/cores/_tmp'

# This path is from within the Solr container
SOLR_TEMP_CORE_PATH_REMOTE = '/etc/opt/solr/_tmp'
```

Reply:
OK, I get it now. Kinda messy. A couple thoughts:

1. I think the variables and comments would make more sense if they reflected the docker images somehow. So, for example, something like SOLR_CORE_IN_DOCKER would be a lot more clear, I think. 

2. But setting that aside, I wonder if it'd be better to start up the docker container for the tests and then destroy it at the end? That'd give a *very* clean work environment and it might be less confusing that the above (maybe). One of the problems the tests have every so often is that the temp core didn't get deleted properly at the end of a test because the test crashed or something. It's a pain to fix this when it happens. If we were using a fresh container that we later deleted, that'd easily fix this problem. What do you think?

Reply:
1. I wanted to keep the variable names generic in case Solr just runs on a different remote machine and not in Docker. If such a scenario is not realistic, naming it `SOLR_TEMP_CORE_DOCKER_PATH` would be fine.

2. In general, I wanted to change as little as possible in your code. But I must admit that creating a Solr core with FS operations is probably not the best way. Isn't possible to create a new core only over the API? See https://stackoverflow.com/questions/21619947/create-new-cores-in-solr-via-http

Reply:
I think we can code up CL under the assumption it'll be using freelawproject/solr going forward, but also, it seems like copying files around is for chumps and we should use the API instead. 

If you see a way of improving the code, please go for it.